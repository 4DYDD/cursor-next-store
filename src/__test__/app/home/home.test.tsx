import Home from "@/app/page";
import { render } from "@testing-library/react";

describe("Home Page", () => {
  it("should render the home page", () => {
    const page = render(<Home />);
    expect(page).toMatchSnapshot();
  });
});
