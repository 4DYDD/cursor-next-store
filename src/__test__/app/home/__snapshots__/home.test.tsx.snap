// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render the home page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="min-h-screen bg-gray-100"
      >
        <div
          class="container mx-auto px-4 py-8"
        >
          <h1
            class="text-4xl font-extrabold text-center text-gray-800 mb-8"
          >
            Selamat Datang di Tokoku
          </h1>
          <div
            class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6"
          >
            <div
              class="bg-white p-6 rounded-lg shadow-md"
            >
              <div
                class="flex items-center mb-4"
              >
                <svg
                  class="w-8 h-8 text-green-500 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 13l4 4L19 7"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <h2
                  class="text-xl font-bold text-gray-700"
                >
                  Produk Terbaik
                </h2>
              </div>
              <p
                class="text-gray-600 font-semibold"
              >
                Kami menyediakan produk-produk berkualitas tinggi dengan harga terbaik.
              </p>
            </div>
            <div
              class="bg-white p-6 rounded-lg shadow-md"
            >
              <div
                class="flex items-center mb-4"
              >
                <svg
                  class="w-8 h-8 text-teal-500 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <h2
                  class="text-xl font-bold text-gray-700"
                >
                  Pengiriman Cepat
                </h2>
              </div>
              <p
                class="text-gray-600 font-semibold"
              >
                Pesanan Anda akan dikirim dengan cepat dan aman ke alamat tujuan.
              </p>
            </div>
            <div
              class="bg-white p-6 rounded-lg shadow-md"
            >
              <div
                class="flex items-center mb-4"
              >
                <svg
                  class="w-8 h-8 text-blue-500 mr-2"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 12l2 2 4-4m5.618-4.016A11.955 11.955 0 0112 2.944a11.955 11.955 0 01-8.618 3.04A12.02 12.02 0 003 9c0 5.591 3.824 10.29 9 11.622 5.176-1.332 9-6.03 9-11.622 0-1.042-.133-2.052-.382-3.016z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <h2
                  class="text-xl font-bold text-gray-700"
                >
                  Layanan 24/7
                </h2>
              </div>
              <p
                class="text-gray-600 font-semibold"
              >
                Tim customer service kami siap membantu Anda kapan saja.
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="min-h-screen bg-gray-100"
    >
      <div
        class="container mx-auto px-4 py-8"
      >
        <h1
          class="text-4xl font-extrabold text-center text-gray-800 mb-8"
        >
          Selamat Datang di Tokoku
        </h1>
        <div
          class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6"
        >
          <div
            class="bg-white p-6 rounded-lg shadow-md"
          >
            <div
              class="flex items-center mb-4"
            >
              <svg
                class="w-8 h-8 text-green-500 mr-2"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 13l4 4L19 7"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
              <h2
                class="text-xl font-bold text-gray-700"
              >
                Produk Terbaik
              </h2>
            </div>
            <p
              class="text-gray-600 font-semibold"
            >
              Kami menyediakan produk-produk berkualitas tinggi dengan harga terbaik.
            </p>
          </div>
          <div
            class="bg-white p-6 rounded-lg shadow-md"
          >
            <div
              class="flex items-center mb-4"
            >
              <svg
                class="w-8 h-8 text-teal-500 mr-2"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
              <h2
                class="text-xl font-bold text-gray-700"
              >
                Pengiriman Cepat
              </h2>
            </div>
            <p
              class="text-gray-600 font-semibold"
            >
              Pesanan Anda akan dikirim dengan cepat dan aman ke alamat tujuan.
            </p>
          </div>
          <div
            class="bg-white p-6 rounded-lg shadow-md"
          >
            <div
              class="flex items-center mb-4"
            >
              <svg
                class="w-8 h-8 text-blue-500 mr-2"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 12l2 2 4-4m5.618-4.016A11.955 11.955 0 0112 2.944a11.955 11.955 0 01-8.618 3.04A12.02 12.02 0 003 9c0 5.591 3.824 10.29 9 11.622 5.176-1.332 9-6.03 9-11.622 0-1.042-.133-2.052-.382-3.016z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
              <h2
                class="text-xl font-bold text-gray-700"
              >
                Layanan 24/7
              </h2>
            </div>
            <p
              class="text-gray-600 font-semibold"
            >
              Tim customer service kami siap membantu Anda kapan saja.
            </p>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
